nix/sourcebot.nix

```diff
@@ -0,0 +1,111 @@
+{
+  pkgs,
+  lib,
+}:
+pkgs.stdenv.mkDerivation (finalAttrs: {
+  name = "sourcebot";
+
+  src = ../.;
+
+  buildInputs = with pkgs; [
+    nodejs
+    yarn
+    redis
+    openssl
+  ];
+  buildPhase = ''
+    runHook preBuild
+    ${lib.getExe pkgs.yarn-berry} workspaces foreach -R --from '{@sourcebot/schemas,@sourcebot/error,@sourcebot/db}' run build
+    ${lib.getExe pkgs.yarn-berry} build:deps
+    ${lib.getExe pkgs.yarn-berry} build
+    runHook postBuild
+  '';
+  missingHashes = ./missing-hashes.json;
+  offlineCache = pkgs.yarn-berry.fetchYarnBerryDeps {
+    inherit (finalAttrs) src missingHashes;
+    hash = "sha256-mM268PS5jgT+glNRtjNMLV936Z/Xgd16YavADWUiSmE=";
```

Comment:
@msukkari , this line would need to be updated every commit or release: https://github.com/NixOS/nixpkgs/blob/master/doc%2Flanguages-frameworks%2Fjavascript.section.md#yarn-berry_xfetchyarnberrydeps-javascript-fetchyarnberrydeps

I suggest to go with release based update, and I will update the doc on how to pick specific version.